### PR TITLE
use secp256k1 library for signing

### PR DESCRIPTION
**Pull request by pedrouid**
## Checklist:

- [ ] I am making this PR against staging, not master
- [ ] My code follows the code style of this project.
- [ ] I have described any backwards-incompatibility implications above.
- [ ] I have highlighted which parts of the code should be reviewed most carefully.
- [ ] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

File: modules/client/src/lib/utils.ts [hunk 1]
```diff
@@ -86,3 +86,19 @@ export const createRandom32ByteHexString = (): string => {
...
 export const createPaymentId = createRandom32ByteHexString;
 export const createPreImage = createRandom32ByteHexString;
+
+export function removeTrailing0x(str: string) {
+  if (str.startsWith("0x")) {
+    return str.substring(2);
+  } else {
+    return str;
+  }
+}
```
Comment (bohendo) [inline modules/client/src/lib/utils.ts]: 1. This should be called remove**Leading**0x
2. This prob doesn't need to be a separate function bc we can just do `.replace(/^0x/, '')`

Comment (LayneHaber) [reply modules/client/src/lib/utils.ts]: Good catch -- completely missed this. I don't mind having an extra function for it, but we should update the name

Comment (pedrouid) [reply modules/client/src/lib/utils.ts]: Makes sense, I've renamed and refactored these hex utils in commit https://github.com/ConnextProject/indra/pull/815/commits/2cd997f308261f25d2912b592d19a2b958cd6169

**Comment by pedrouid** [comment]
This was breaking with ConnextBenchmarking and even simply loading the DaiCard, I'm unsure how this even passed all the tests. We should investigate the case where signing breaks but all the tests are passing.

**Pull request closed without merging**